common/src/unifyfs_keyval.c

```diff
@@ -543,8 +544,8 @@ static int unifyfs_fskv_init(unifyfs_cfg_t* cfg)
         }
 
         // find or create rank-specific subdir
-        snprintf(sharedfs_rank_kvdir, sizeof(sharedfs_rank_kvdir), "%s/%d",
-                 sharedfs_kvdir, kv_myrank);
+        assert(snprintf(sharedfs_rank_kvdir, sizeof(sharedfs_rank_kvdir), "%s/%d",
```

Comment:
GCC gives a warning basically saying 'you could be truncating, so you should be checking the return code".  However is super-super unlikely we're going to truncate the string (which is why the old code didn't check it), so the assert() was a convenient way to check it to get rid of the warning.  You're right though that it's going to break if they #define NODEBUG, so that's not gonna work.   I'll need to change all the asserts() to something else.

Reply:
Fixed latest push